src/PriceOracle.ts

```diff
@@ -100,114 +110,74 @@ export async function set_whitelisted_prices(
   const tokenData =
     chainId === 10 ? OPTIMISM_WHITELISTED_TOKENS : BASE_WHITELISTED_TOKENS;
 
-  // Get prices from oracle
+  // Get prices from oracle and filter if token is not created yet
   const addresses = tokenData
     .filter((token) => token.createdBlock <= blockNumber)
     .map((token) => token.address);
 
-  if (addresses.length === 0) return;
-  const units = tokenData.map((token) => token.unit as validUnit);
-
-  // Initialize the token price cache
-  const tokenPriceCache = Cache.init(CacheCategory.TokenPrice, chainId);
-
-  // Check cache for existing prices
-  const cachedPrices: string[] = [];
-  const addressesToFetch: string[] = [];
-  addresses.forEach((address, index) => {
-    const cacheKey = `${chainId}_${address}_${blockNumber}`;
-    const cachedPrice = tokenPriceCache.read(cacheKey);
-    if (
-      cachedPrice &&
-      typeof cachedPrice === "object" &&
-      "price" in cachedPrice
-    ) {
-      cachedPrices[index] = cachedPrice.price;
-    } else {
-      addressesToFetch.push(address);
-    }
-  });
+  if (addresses.length === 0) return; 
 
-  // Fetch prices for addresses not in cache
-  let fetchedPrices: string[] = [];
-  if (addressesToFetch.length > 0) {
-    fetchedPrices = await read_prices(addressesToFetch, chainId, blockNumber);
-  }
+  const tokenPriceCache = Cache.init(CacheCategory.TokenPrices, chainId);
 
-  // Combine cached and fetched prices
-  const prices = addresses.map((address, index) => {
-    if (cachedPrices[index]) {
-      return cachedPrices[index];
-    } else {
-      const fetchedIndex = addressesToFetch.indexOf(address);
-      return fetchedPrices[fetchedIndex];
-    }
-  });
+  // Check cache for existing prices list by hashing list of addresses.
+  // If there is any new addresses, we will need to fetch new prices.
+  const addressHash: string = hashAddresses(addresses);
+  const cacheKey = `${chainId}_${addressHash}_${blockNumber}`;
```

Comment:
Here is the bigger change. Since the fetch call happens on any missing address, we might as well remove all the address price checking logic since we have to make the RPC call if there's any new address.